apps/quanta/app/loaders/content.ts

```diff
@@ -0,0 +1,64 @@
+import type { Route } from '../+types/root';
+import { data } from 'react-router';
+import PloneClient from '@plone/client';
+import config from '@plone/registry';
+import type { Content } from '@plone/types';
+
+const expand = ['navroot', 'breadcrumbs', 'navigation'];
+
+/**
+ * The definitive flattenToAppURL function
+ * Flattens all the URLs in the response to the current app URL
+ * This could be a potential use case for the upcoming RR7 middleware
+ */
+function flattenToAppURL(data: Content, request: Request) {
+  const currentUrl = new URL(request.url);
+  const baseUrl = `${currentUrl.protocol}//${currentUrl.host}`;
+
+  // Convert data to string to perform replacements
+  let stringData = JSON.stringify(data);
+
+  // Replace all occurrences of backend URLs
+  stringData = stringData.replace(
+    new RegExp(config.settings.apiPath, 'g'),
```

Comment:
Do we really need to use a regexp here? I understand this might be just copied over from the current Volto implementation, but we don't really expect to have regular expressions in `apiPath`, actually the `RegExp` constructor might break for some special characters if they are used in `apiPath`.